pkg/platform/runtime/setup/setup.go

```diff
@@ -401,19 +404,47 @@ func (s *Setup) fetchAndInstallArtifactsFromBuildPlan(installFunc artifactInstal
 
 	// Compute and handle the change summary
 	// runtimeAndBuildtimeArtifacts records all artifacts that will need to be built in order to obtain the runtime.
-	// Disabled due to DX-2033.
-	// Please use this var when we come back to this in the future as we need to make a clear distinction between this
-	// and runtime-only artifacts.
-	// var runtimeAndBuildtimeArtifacts artifact.Map
+	var runtimeAndBuildtimeArtifacts artifact.Map
 	var runtimeArtifacts artifact.Map // Artifacts required for the runtime to function
-	if buildResult.Build != nil {
-		runtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build)
+
+	// If the build is not ready, we need to get the runtime and buildtime closure
+	if !buildResult.BuildReady {
+		runtimeAndBuildtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build, true)
+		if err != nil {
+			return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
+		}
+	}
+
+	var includeBuildtimeClosure bool
+	// If we are installing build dependencies, then buildtime dependencies are also runtime dependencies
+	if strings.EqualFold(os.Getenv(constants.InstallBuildDependencies), "true") {
+		logging.Debug("Installing build dependencies")
+		includeBuildtimeClosure = true
+		if runtimeAndBuildtimeArtifacts == nil {
+			runtimeAndBuildtimeArtifacts, err = buildplan.NewMapFromBuildPlan(buildResult.Build, includeBuildtimeClosure)
+			if err != nil {
+				return nil, nil, errs.Wrap(err, "Failed to create artifact map from build plan")
+			}
+		}
+		runtimeArtifacts = runtimeAndBuildtimeArtifacts
```

Comment:
This is raising a red flag for me. I'd rather we be explicit. I would suggest capturing the runtime and build time closures separately, and combining them for use-cases where that may be needed. Note I am not suggesting you calculate them separately, but you can do a single calculation that captures both.